### PR TITLE
use harmony prepare instead of manual patch

### PR DESCRIPTION
**Pull request by kianzarrin**
Review hint: I have not added any new line of code here. I have only moved/modified several lines. But mostly I have deleted lines.

@sway2020 Sorry to change your code but I think:
1- there is no need to use manual patch when we can just use harmony prepare. This way it patches are more consistent/self-contained and have less lines of code.
2- there is no need to use `DeclaredMethod<TargetDelegate>()` if the target method is not overloaded. A simple `Type.GetMethod()` would do.

latest test build: [https://ci.appveyor.com/api/projects/krzychu124/tmpe/artifacts/TMPE.zip?branch=prepare-patch](TMPE.zip)

### CODE REVIEW

**Comment by kianzarrin** [comment]
Test: I loaded a new city and there was no error. I don't have a city to test if this works though.

File: TLM/TLM/Patch/_External/RTramAIModPatch.cs [hunk 1]
```diff
@@ -1,55 +1,35 @@
...
+        private static Type TargetType => Type.GetType("ReversibleTramAI.SimulationStepPatch1, ReversibleTramAI", false);
 
-        private delegate void TargetDelegate(VehicleAI __instance, ushort vehicleID, ref Vehicle data, Vector3 physicsLodRefPos, ref VehicleInfo ___m_info);
-        private delegate bool TrySpawnDelegate(ushort vehicleID, ref Vehicle vehicleData);
-
-        public static bool ApplyPatch(Harmony harmonyInstance, Type simulationStepPatch1Type) {
-            try {
-                var original = TranspilerUtil.DeclaredMethod<TargetDelegate>(simulationStepPatch1Type, "Prefix");
-                if (original == null) return false;
+        public static bool Prepare() => TargetType != null;
```
Comment (krzychu124) [inline TLM/TLM/Patch/_External/RTramAIModPatch.cs:22]: Interesting, didn't know about that 🤔

**Review by krzychu124 — APPROVED** [review]
I'll test in game later but it looks good so I doubt it won't work

**Comment by kianzarrin** [comment]
@chameleon-tbn Can you please test if this works with the mod?

**Comment by sway2020** [comment]
@kianzarrin no need to say sorry! I am far away from my gaming PC so I can't run the game and test this

**Review by aubergine10 — APPROVED** [review]
👍

**Pull request merged**